lib/sqlalchemy/ext/automap.py

```diff
@@ -572,19 +571,56 @@ def column_reflect(inspector, table, column_info):
 
 
 """  # noqa
+from __future__ import annotations
+
+from typing import Any
+from typing import Callable
+from typing import cast
+from typing import Dict
+from typing import List
+from typing import NoReturn
+from typing import Optional
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
+from typing import Union
+
 from .. import util
+from ..engine.base import Engine
 from ..orm import backref
 from ..orm import declarative_base as _declarative_base
 from ..orm import exc as orm_exc
 from ..orm import interfaces
 from ..orm import relationship
+from ..orm.base import RelationshipDirection
```

Comment:
**mike bayer** (zzzeek) wrote:

all the extra imports here that are for typing only should go under TYPE_CHECKING

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4271